compiler/rustc_middle/src/hir/mod.rs

```diff
@@ -157,14 +157,19 @@ pub fn provide(providers: &mut Providers) {
         let hir_id = hir.local_def_id_to_hir_id(def_id);
         if let Some(body_id) = hir.maybe_body_owned_by(def_id) {
             tcx.arena.alloc_from_iter(hir.body_param_names(body_id))
-        } else if let Node::TraitItem(&TraitItem {
-            kind: TraitItemKind::Fn(_, TraitFn::Required(idents)),
-            ..
-        }) = hir.get(hir_id)
-        {
-            tcx.arena.alloc_slice(idents)
         } else {
-            span_bug!(hir.span(hir_id), "fn_arg_names: unexpected item {:?}", id);
+            match hir.get(hir_id) {
+                Node::TraitItem(&TraitItem {
+                    kind: TraitItemKind::Fn(_, TraitFn::Required(idents)),
+                    ..
+                })
+                | Node::ForeignItem(&ForeignItem {
+                    kind: ForeignItemKind::Fn(_, idents, _), ..
+                }) => tcx.arena.alloc_slice(idents),
```

Comment:
```suggestion
                }) => idents,
```
This is already a slice with the correct lifetime.
Nit: I would have kept this an if-let with an OR-pattern.
r=me either way.